cmake/CPM.cmake

```diff
@@ -650,6 +650,18 @@ function(CPMAddPackage)
     list(APPEND CPM_ARGS_UNPARSED_ARGUMENTS DOWNLOAD_COMMAND ${CPM_ARGS_DOWNLOAD_COMMAND})
   elseif(DEFINED CPM_ARGS_SOURCE_DIR)
     list(APPEND CPM_ARGS_UNPARSED_ARGUMENTS SOURCE_DIR ${CPM_ARGS_SOURCE_DIR})
+    if(NOT IS_ABSOLUTE ${CPM_ARGS_SOURCE_DIR})
+      # Expand `CPM_ARGS_SOURCE_DIR` relative path. This is important because EXISTS doesn't work for
+      # relative paths.
+      get_filename_component(source_directory ${CPM_ARGS_SOURCE_DIR} REALPATH BASE_DIR ${CMAKE_BINARY_DIR})
```

Comment:
I think it should be `CMAKE_CURRENT_BINARY_DIR`, as according to the [CMake docs](https://cmake.org/cmake/help/latest/module/ExternalProject.html):

> Relative paths are interpreted with respect to [CMAKE_CURRENT_BINARY_DIR](https://cmake.org/cmake/help/latest/variable/CMAKE_CURRENT_BINARY_DIR.html#variable:CMAKE_CURRENT_BINARY_DIR) at the point where ExternalProject_Add() is called.

Reply:
Thanks, this is fixed now.